data/binding/binding.go

```diff
@@ -0,0 +1,58 @@
+//go:generate go run gen.go
+
+package binding
+
+// DataItem is the base interface for all bindable data items.
+type DataItem interface {
+	AddListener(DataItemListener)
+	RemoveListener(DataItemListener)
+}
+
+// DataItemListener is any object that can register for changes in a bindable DataItem.
+// See NewDataItemListener to define a new listener using just an inline function.
+type DataItemListener interface {
+	DataChanged(DataItem)
+}
+
+// NewDataItemListener is a helper function that creates a new listener type from a simple callback function.
+func NewDataItemListener(fn func(DataItem)) DataItemListener {
+	return &listener{fn}
+}
+
+type listener struct {
+	callback func(DataItem)
+}
+
+func (l *listener) DataChanged(i DataItem) {
+	l.callback(i)
+}
+
+type base struct {
+	listeners []DataItemListener
+}
+
+// AddListener allows a data listener to be informed of changes to this item.
+func (b *base) AddListener(l DataItemListener) {
+	b.listeners = append(b.listeners, l)
```

Comment:
New listeners should be triggered with the current state: https://github.com/fyne-io/fyne/pull/804/files#diff-3f981e7a9d32b5147a37811648cc9f675d6f4519a1abd33797711994659da8d2R55

Reply:
That is cool for sure, but is it the expectation of a developer?
Normally listeners are triggers of change - when we are starting with a provided value (i.e. BindFloat(myVal) ) the developer already knows.
Not disagreeing, just wanted to make sure we considered this angle.

Reply:
After chat on Slack it was agree that adding this with some clear docs that explain it would be sufficient.
Resolved.